File: src/uimd/dialogs/message_box_yes_no_cancel.py
"""Standard Yes/No/Cancel message box."""

try:
    from uimd.dialogs.message_box import MessageBoxBehavior
    from uimd.dialogs.message_box_yes_no_cancel_ui import MessageBoxYesNoCancelUI
except ImportError:
    from message_box import MessageBoxBehavior
    from message_box_yes_no_cancel_ui import MessageBoxYesNoCancelUI


class MessageBoxYesNoCancel(MessageBoxBehavior, MessageBoxYesNoCancelUI):
    def __init__(self, header="Message", message="", on_close=None):
        super().__init__()
        self._configure_message_box(
            header=header,
            message=message,
            on_close=on_close,
            result_by_button={
                "yes_btn": "yes",
                "no_btn": "no",
                "cancel_btn": "cancel",
            },
            default_result="cancel",
        )

File: java/src/main/java/uimd/MessageBoxYesNoCancel.java
package uimd;

import java.util.Map;
import java.util.function.Consumer;

public final class MessageBoxYesNoCancel extends MessageBoxYesNoCancelUI
{
    private final MessageBoxBehavior behavior;

    public MessageBoxYesNoCancel()
    {
        this("Message", "", null);
    }

    public MessageBoxYesNoCancel(String header, String message)
    {
        this(header, message, null);
    }

    public MessageBoxYesNoCancel(String header, String message, Consumer<String> onClose)
    {
        behavior = new MessageBoxBehavior(
            dialog_header,
            this.message,
            header,
            message,
            onClose,
            Map.of("yes_btn", "yes", "no_btn", "no", "cancel_btn", "cancel"),
            "cancel");
    }

    public String result()
    {
        return behavior.result();
    }

    public boolean closed()
    {
        return behavior.closed();
    }

    public int run(String[] arguments)
    {
        return GeneratedWindowRuntime.runGeneratedWindow(this, runtimeOptions(), arguments);
    }

    @Override
    public GeneratedWindowRuntimeOptions runtimeOptions()
    {
        return behavior.configure(super.runtimeOptions());
    }

    @Override
    protected void onYesBtnClick()
    {
        behavior.handleButton("yes_btn");
    }

    @Override
    protected void onNoBtnClick()
    {
        behavior.handleButton("no_btn");
    }

    @Override
    protected void onCancelBtnClick()
    {
        behavior.handleButton("cancel_btn");
    }
}
